presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -447,6 +444,26 @@ public PlanWithProperties visitSort(SortNode node, PreferredProperties preferred
                 }
             }
 
+            if (isDistributedSortEnabled(session)) {
```

Comment:
> Should we also add && !child.getProperties().isSingleNode()?

Large single node stream might not fit within node memory. If distributed sort is enabled, than such large stream is redistributed across nodes (there was `redistribute_sort` flag enabled by default, which I removed and added unconditional round robin exchange here) so that entire cluster memory can be utilized for sorting.

> Alternatively, it might be clearer to complete processing of single-node streams before coming here by adding

Both single node and multi-node streams can be processed with distributed sort, so there is no entirely different code path for single node streams.